openedx/core/djangoapps/user_api/management/commands/sync_hubspot_contacts.py

```diff
@@ -0,0 +1,241 @@
+"""
+Management command to sync platform users with hubspot
+./manage.py lms sync_hubspot_contacts
+./manage.py lms sync_hubspot_contacts --initial-sync-days=7 --batch-size=20
+"""
+
+import json
+import traceback
+from datetime import datetime, timedelta
+
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand, CommandError
+from django.utils.html import escapejs
+
+from edx_rest_api_client.client import EdxRestApiClient
+from slumber.exceptions import HttpClientError, HttpServerError
+
+from openedx.core.djangoapps.site_configuration.models import SiteConfiguration
+from student.models import UserAttribute, UserProfile
+from util.query import use_read_replica_if_available
+
+
+HUBSPOT_API_BASE_URL = 'https://api.hubapi.com'
+
+
+class Command(BaseCommand):
+    """
+    Command to create contacts in hubspot for those partner who has enabled hubspot integration.
+    This command is suppose to sync contact with hubspot on daily basis.
+    """
+
+    def _get_hubspot_enabled_sites(self):
+        """
+        Returns: list of site configurations having hubspot integration enabled
+        """
+        site_confs = SiteConfiguration.objects.all()
+        hubspot_sites = [
+            site_conf for site_conf in site_confs
+            if site_conf.get_value('HUBSPOT_API_KEY')
+        ]
+        return hubspot_sites
+
+    def _get_last_synced_contact_email(self, site_conf):
+        """
+        Returns: last synced contact email for given site
+
+        """
+        api_key = site_conf.get_value('HUBSPOT_API_KEY')
+        last_contact_email = None
+        client = EdxRestApiClient('/'.join([HUBSPOT_API_BASE_URL, 'contacts/v1/lists/all/contacts']))
```

Comment:
I don't believe this will work if there are more than 100 contacts as the response will be paginated.

Reply:
This API call is going to return us the most recent contact saved on hubspot. See the `count` parameter we set below which means return just one contact. Here are the details of this API
https://developers.hubspot.com/docs/methods/contacts/get_recently_created_contacts

Reply:
Thanks @ziafazal , i saw the URL there and thought we were calling the `/contacts` endpoint but I see below where we're calling `recent` with slumber.  👍